File: app/src/main/java/com/droidrank/tictactoe/MainContract.java
package com.droidrank.tictactoe;

import com.droidrank.tictactoe.base.BasePresenter;
import com.droidrank.tictactoe.domain.Board;

public interface MainContract {
    interface Presenter extends BasePresenter<View> {
        void onBlockClick(int x, int y);
        void onRestartClick();
        void restart();
    }

    interface View {
        String getPlayer1Symbol();
        String getPlayer2Symbol();
        void render(Board board);
        void showRestartDialog();
    }
}

File: app/src/main/java/com/droidrank/tictactoe/MainContract.kt
package com.droidrank.tictactoe

import com.droidrank.tictactoe.base.BasePresenter
import com.droidrank.tictactoe.domain.Board

interface MainContract {

    interface Presenter : BasePresenter<View> {
        fun onBlockClick(x: Int, y: Int)
        fun onRestartClick()
        fun restart()
    }

    interface View {
        val player1Symbol: String
        val player2Symbol: String
        fun render(board: Board)
        fun showRestartDialog()
    }

}
